torchvision/csrc/io/image/cuda/encode_jpeg_cuda.cpp

```diff
@@ -0,0 +1,236 @@
+#include <torch/nn/functional.h>
+#include "c10/core/ScalarType.h"
+#include "encode_decode_jpeg_cuda.h"
+#include "torch/types.h"
+
+#include <ATen/ATen.h>
+#include <iostream>
+#include <memory>
+
+#if NVJPEG_FOUND
+#include <ATen/cuda/CUDAContext.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <nvjpeg.h>
+
+nvjpegHandle_t nvjpeg_handle = nullptr;
+std::once_flag nvjpeg_handle_creation_flag;
+
+#endif
+#include <string>
+
+namespace vision {
+namespace image {
+
+#if !NVJPEG_FOUND
+
+std::vector<torch::Tensor> encode_jpeg_cuda(
+    const std::vector<torch::Tensor>& images,
+    const int64_t quality) {
+  TORCH_CHECK(
+      false, "decode_jpeg_cuda: torchvision not compiled with nvJPEG support");
+}
+
+#else
+
+void nvjpeg_init() {
```

Comment:
This probably doesn't matter too much but `nvjpeg_init()` is declared in `encode_decode_jpeg_cuda.h` no matter what `NVJPEG_FOUND` is, but it is only defined here if `NVJPEG_FOUND` is defined.